src/lib/tasks/Login.ts

```diff
@@ -296,16 +283,27 @@ export class LoginTask extends Task<LoginTaskParams> {
     });
   }
 
-  async #handleSpecForm(): Promise<void> {
-    const { log } = this;
+  #needSpec(): 'login.live.com' | false {
     if (!this.page?.ref) {
-      return;
+      return false;
     }
 
     const currentUrl = this.page.ref.url();
+    if (currentUrl.startsWith('https://login.live.com')) {
+      return 'login.live.com';
+    }
+
+    return false;
+  }
+
+  async #handleSpecForm({ spec }: { spec: 'login.live.com' }): Promise<void> {
+    const { log } = this;
+    if (!this.page?.ref) {
+      return;
+    }
 
     // Spec for Microsoft SSO
-    if (currentUrl.startsWith('https://login.live.com')) {
+    if (spec === 'login.live.com') {
```

Comment:
Naming: I had to read until the comment above to understand that "Spec" is not for "specification" but for "special case" right?
Could you rename everywhere? :pray: